### PR TITLE
Implement Callstack class at dag/utils/callstack.py

### PR DESCRIPTION
**Pull request by Chiiizzzy**
Implement Callstack class at dag/utils/callstack.py
#12

### CODE REVIEW

File: towhee/dag/utils/callstack.py [hunk 1]
```diff
@@ -11,55 +11,110 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+import inspect
+from typing import List
+import hashlib
+import logging
```
Comment (junjiejiangjjj) [inline towhee/dag/utils/callstack.py:17]: logging is unused

File: towhee/dag/utils/callstack.py [hunk 2]
```diff
@@ -11,55 +11,110 @@
...
-        Get the number of frames
+        Get the number of frames.
+
+        Returns:
+            The size of current stack.
         """
-        raise NotImplementedError
+        return self.size
 
     def find_func(self, func_name: str) -> int:
```
Comment (junjiejiangjjj) [inline towhee/dag/utils/callstack.py:48]: If function will return None, use Optional[int]

File: towhee/tests/dag/utils/test_callstack.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    block the frames produced by the unittest. We do not care about_ the process of
+    tests, we want to focus only on the architectures that we designed, which in this
+    case are `test_xxx`, `generate_callstack` and `collect_frames`. In this way, the 
+    structure and depth of the callstack is stable and predictable.
+    For convenience, we collect and returns the frames inside the callsatck in the
+    the form of list.
+    """
+
+    def test_init(self):
+        # create a callstack contains 3 frames
```
Comment (junjiejiangjjj) [inline towhee/tests/dag/utils/test_callstack.py:43]: Show the function stack in the comment

File: towhee/tests/dag/utils/test_callstack.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        s1 = generate_callstack()
```
Comment (junjiejiangjjj) [inline towhee/tests/dag/utils/test_callstack.py:44]: Put all s1 tests together, and s2, s3, s4

**Comment by Chiiizzzy** [comment]
git problems fixed and rebased @GuoRentong @junjiejiangjjj

**Review by junjiejiangjjj — APPROVED** [review]
/approve
/lgtm

**Pull request merged**